### ISSUE TITLE
Android app: per-subscription / per-priority notification sound

### ISSUE THREAD

**Issue by binwiederhier** [binwiederhier/ntfy#6]

**Comment by Mikaela** [comment]
This would also be nice on iOS as I tend to default to silent so most applications won't bother me unnecessarily

**Comment by inconis** [comment]
On Android devices will be helpful the option to override system settings regarding noise. 
In this way is possible to create an SOS channel to receive alarms in every condition.
A big thank to the developers of this project that I found so useful.

**Comment by inconis** [comment]
Hi, thanks for your answer. Probably I explained in a bad way what I meant. With the app, "WirePusher" is possible to redirect the stream to "alarm stream" in this way if I put my phone in silent mode I can still receive the sound of the notification. 
This function can be really useful for example to receive some "urgent" notifications.

**Comment by binwiederhier** [comment]
See also #404

**Comment by doits** [comment]
Maybe this could be implemented by creating custom notification channels when enabled for a subscription, like toggling a setting "Custom notification channels for this subscription".

Currently there are 5 default notification channels (min to max). If the setting was toggled for a specific subscription, it could add 5 more notification channels for the specific channel. Then you could set the notification sound etc. within the android app settings, even override DND, like it is now possible for the 5 default notification channels.

Another way (or in addition) maybe you could create custom notification channels globally in the ntfy app settings and then assign a subscription to that notification channel. E.g. in settings you create a channel called "X" and ntfy would create 5 notification channels for "X", one for each priority (min to max). This way you could put multiple subscriptions into the same custom notification channel (not sure if this is needed).

**Comment by binwiederhier** [comment]
@doits This is an incredibly smart idea. And I love that you even provided an implementation for it. I would even go so far as to just create a notification channel for every subscription by default. I'm gonna have a look at the code today if I can.

**Comment by binwiederhier** [comment]
Okay, upon thinking about it, having a custom channel for each subscription by default is not going to work, because then the priority-channels would be irrelevant. It's kind of a tricky one. I think _if_ we were to make this a thing, it'd have to be opt-in like you suggested:

With these channels:
```
- Subscription service
- Min priority
- Low priority
- Default priority
- High priority
- Max priority
- ntfy.sh/mytopic (if opted in)
- ntfy.sh/othertopic (if opted in)
```

I think adding multiple "per-priority channels" for each topic is overkill.

**Comment by binwiederhier** [comment]
Alright ... So I should have thought about the whole thing and maybe even tried it before commenting. I do think that maybe your approach is actually the correct one. Otherwise you'd essentially take away the ability to have low/high-priority messages when this is enabled.

So what you implemented is this:

```
Background service
- Subscription service

Priority channels
- Min priority
- Low priority
- Default priority
- High priority
- Max priority

ntfy.sh/mytopic
- Min priority
- Low priority
- Default priority
- High priority
- Max priority
```

While this seems like the only approach that achieves "infinite flexibility", it also seems like absolute configuration overkill. Say I have 5 subscriptions, that means I now have 25 channels if I enable the feature on a subscription. Doing a simple thing, like changing the notification sound, now becomes an awful chore.

**Comment by doits** [comment]
Thanks for your review and your feedback. Your last comment is correct about I implemented it.

I also thought about only adding one channel per subscription (ignoring priority), but I felt this was an important feature of ntfy which should always be present.

While writing a response I thought about different variations and finally asked myself the question: What use case is solved by this? I think most of users want to change the notification sound for a subscription, not having to change the sound of 5 priorities (like you said). Changing the notification for each priority also has it's use case, but probably to fewer users.

So maybe this way is cleaner:

You have the setting ...

```
Custom notifications for this subscription
```

... and when you enable it, it creates one extra notification channel:

```
ntfy.sh/mytopic
- All priorities
```

So by default, all priorities of this subscription go into this one channel.

In addition, it reveals 5 more settings:

```
Custom notifications for a specific priority:
- Min priority
- Low priority
- Default priority
- High priority
- Max priority
```

All are OFF by default. Each toggle creates an extra notification channel just for this subscription, so for example:

```
Custom notifications for a specific priority:
- Min priority: OFF
- Low priority: OFF
- Default priority: OFF
- High priority: ON
- Max priority: ON
```

... results in ...

```
ntfy.sh/mytopic
- High priority
- Max priority
- Other priorities
```

So you can add specific notification channels per priority. If not, they go into the "Other priorities" channel of the subscription.

Finally, if you create a channel for each priority:

```
Custom notifications for a specific priority:
- Min priority: ON
- Low priority: ON
- Default priority: ON
- High priority: ON
- Max priority: ON
```

... it would result in channels like the current implementation ...

```
ntfy.sh/mytopic
- Min priority
- Low priority
- Default priority
- High priority
- Max priority
```

(no need for the "Other priorities" notification channel).

What do you think? Still too complicated?

**Comment by doits** [comment]
A simplification could be to have the user only choose between three states:

```
Custom notifications for this subscription:
- Off (No custom notifications)
- On (All notifications have the same custom notification channel, priority not configurable --> one channel)
- Per priority (One notification channel for each priority --> 5 channels)
```

So in essence you could only choose to have all 5 priority channels or none, which would make the settings dialog simpler (only one list setting) and sacrifice only a little bit flexibility.

**Comment by binwiederhier** [comment]
Your suggestion in the last comment sounds fantastic and tackles all the issues I was worried about. However, option B is so incredibly counter-intiutive ("why does my min priority message still ring?", "why does my max priority message not vibrate like a maniac?") that it doesn't seem like it should exist. 

So after all this discussion, **I feel like what you have seems to be the right approach. Let's go with that!** I still have not reviewed the code, but I will probably tonight.

**Comment by binwiederhier** [comment]
FYI I still haven't reviewed it, and I'm too tired to do it tonight. I got held up by all that "Google rejecting the app" business (see #531 ...).

**Comment by doits** [comment]
No problem, take the time you need. There's no push from my side.

**Comment by binwiederhier** [comment]
I merged https://github.com/binwiederhier/ntfy-android/pull/49 and it'll be in the next release. Thank you @doits for all your work.
